code/modules/medical/pathology/pathogen_symptoms.dm

```diff
@@ -31,46 +30,69 @@ datum/pathogeneffects
+						SPAWN_DBG(rand(0.5,2) SECONDS)
+							H.show_message("Pretty catchy tune...")
+							H.emote("snap") // consider yourself lucky I haven't implemented snap infection yet, human
+
+	// creates an infective cloud
+	// this should give people better feedback about how be infected and how to avoid it
+	proc/infect_cloud(var/mob/M as mob, var/datum/pathogen/origin, var/amount = 5)
+		var/turf/T = get_turf(M)
+		var/datum/reagent/blood/pathogen/Q = new /datum/reagent/blood/pathogen()
+		var/datum/reagents/R = new /datum/reagents(amount)
+		Q.volume = amount
+		Q.pathogens += origin.pathogen_uid
+		Q.pathogens[origin.pathogen_uid] = origin
+		R.reagent_list += "pathogen"
+		R.reagent_list["pathogen"] = Q
+		Q.holder = R
+		R.update_total()
+		T.fluid_react(R, amount, airborne = 1)
+
+	// creates an infective puddle
+	// this should give people better feedback about how be infected and how to avoid it
+	proc/infect_puddle(var/mob/M as mob, var/datum/pathogen/origin, var/amount = 5)
+		var/turf/T = get_turf(M)
+		var/datum/reagent/blood/pathogen/Q = new /datum/reagent/blood/pathogen()
+		var/datum/reagents/R = new /datum/reagents(amount)
+		Q.volume = amount
+		Q.pathogens += origin.pathogen_uid
+		Q.pathogens[origin.pathogen_uid] = origin
+		R.reagent_list += "pathogen"
+		R.reagent_list["pathogen"] = Q
+		Q.holder = R
+		R.update_total()
+		R.trans_to(T, amount)
 
 	// infect_direct(mob, datum/pathogen) : void
 	// This is the proc that handles direct transmission of the pathogen from one mob to another. This should be called in particular infection scenarios. For example, a sweating person
 	// gets his bodily fluids onto another when they directly disarm, punch, or grab a person.
 	// For INFECT_TOUCH diseases this is automatically called on a successful disarm, punch or grab. When overriding any of these events, use ..() to keep this behaviour.
 	// OVERRIDE: Generally, you do not need to override this.
 	proc/infect_direct(var/mob/target as mob, var/datum/pathogen/origin, contact_type = "touch")
-		var/permeability = get_permeability_score(target)
-		if (permeability < src.permeability_score)
-			return 0
-		if (prob(permeability * infection_coefficient))
-			if (target.infected(origin))
-				if (infect_message)
-					target.show_message(infect_message)
-				logTheThing("pathology", origin.infected, target, "infects %target% with [origin.name] due to symptom [name] through direct contact ([contact_type]).")
-				return 1
+		if (infect_attempt_message)
+			target.show_message(infect_attempt_message)
+			if(istype(target, /mob/living/carbon/human))
+				var/mob/living/carbon/human/H = target
+				if(prob(100-H.get_disease_protection()))
+					if (target.infected(origin))
+						if (infect_message)
+							target.show_message(infect_message)
+						logTheThing("pathology", origin.infected, target, "infects %target% with [origin.name] due to symptom [name] through direct contact ([contact_type]).")
+						return 1
```

Comment:
As you said in discorb, this bit doesn't infect if there's no attempt message :snelterfrerg: